File: regexlib/2021-5-15/python_re2_test_file/regexlib_7018.py
# 7018
# ^([a-zA-Z](?:(?:(?:\w[\.\_]?)*)\w)+)([a-zA-Z0-9])$
# EXPONENT
# nums:5
# EXPONENT AttackString:"a"+"_"*32+"!1 __EOA(iii)"

import re2 as re
from time import perf_counter

regex = """^([a-zA-Z](?:(?:(?:\w[\.\_]?)*)\w)+)([a-zA-Z0-9])$"""
REGEX = re.compile(regex)
for i in range(0, 150000):
    ATTACK = "a" + "_" * i * 1 + "!1 __EOA(iii)"
    LEN = len(ATTACK)
    BEGIN = perf_counter()
    m = REGEX.search(ATTACK)
    # m = REGEX.match(ATTACK)
    DURATION = perf_counter() - BEGIN
    print(f"{i *1}: took {DURATION} seconds!")

File: regexlib/2021-5-15/javascript_test_file/regexlib_7018.js
/* 7018
 * ^([a-zA-Z](?:(?:(?:\w[\.\_]?)*)\w)+)([a-zA-Z0-9])$
 * EXPONENT
 * nums:3
 * EXPONENT AttackString:"a"+"_"*32+"!1 __EOA(iii)"
 */
var REGEX = /^([a-zA-Z](?:(?:(?:\w[\.\_]?)*)\w)+)([a-zA-Z0-9])$/
for(var i = 1; i <= 5000000; i++) {
    var time = Date.now();
    var attack_str = 'a' + '_'.repeat(i*1) + '!1 __EOA(iii)'
    REGEX.exec(attack_str)
    // REGEX.test(attack_str);
    // attack_str.search(REGEX)
    // attack_str.match(REGEX)
    var time_cost = Date.now() - time;
    console.log(i * 1 + ": " + time_cost+" ms")
}